pkg/unshare/unshare_linux.go

```diff
@@ -387,10 +388,39 @@ const (
 	UsernsEnvName = "_CONTAINERS_USERNS_CONFIGURED"
 )
 
+// hasFullUsersMappings checks whether the current user namespace has all the IDs mapped.
+func hasFullUsersMappings() (bool, error) {
+	content, err := ioutil.ReadFile("/proc/self/uid_map")
+	if err != nil {
+		return false, err
+	}
+	// if the uid_map contains 4294967295, the entire IDs space is available in the
+	// user namespace, so it is likely the initial user namespace.
+	return bytes.Contains(content, []byte("4294967295")), nil
```

Comment:
Can it happen that that that single ID is mapped but not the others?

Reply:
The kernel rejects attempts to create mappings where either start point is 4294967295 with EINVAL.

Reply:
Oh, right with you there.  I had to look it up.